lib/utilities/aclUtils.js

```diff
@@ -103,8 +103,8 @@ aclUtils.parseAclXml = function parseAclXml(toBeParsed, log, next) {
         }
         if (!result.AccessControlPolicy
                 || !result.AccessControlPolicy.AccessControlList
-                || !result.AccessControlPolicy
-                .AccessControlList[0].Grant) {
+                || (result.AccessControlPolicy.AccessControlList[0] !== '' &&
+                  !result.AccessControlPolicy.AccessControlList[0].Grant)) {
```

Comment:
Are there multiple ACLs for a single bucket/object? In that case, what if the first ACL is OK but the rest is not?